Meadowlark-js/backends/meadowlark-opensearch-backend/test/integration/QueryOpensearch.test.ts

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import {
+  ResourceInfo,
+  UpsertRequest,
+  NoDocumentInfo,
+  newSecurity,
+  UpsertResult,
+  DeleteRequest,
+  DeleteResult,
+  PaginationParameters,
+  QueryRequest,
+} from '@edfi/meadowlark-core';
+import { Client } from '@opensearch-project/opensearch/.';
+import { getNewClient } from '../../src/repository/Db';
+import { queryDocuments } from '../../src/repository/QueryOpensearch';
+import { afterUpsertDocument, afterDeleteDocumentById } from '../../src/repository/UpdateOpensearch';
+import { setupOpenSearch, teardownOpenSearch } from '../setup/OpenSearchSetup';
+
+jest.setTimeout(120_000);
+
+const resourceInfo: ResourceInfo = {
+  projectName: 'ed-fi',
+  resourceName: 'student',
+  isDescriptor: false,
+  resourceVersion: '3.3.1-b',
+};
+
+const id = '1234a-5678b';
+
+const newUpsertRequest: UpsertRequest = {
+  id,
+  resourceInfo,
+  documentInfo: NoDocumentInfo,
+  edfiDoc: {},
+  validate: false,
+  security: { ...newSecurity() },
+  traceId: 'traceId',
+};
+
+const setupQueryRequest = (queryParameters: any, paginationParameters: PaginationParameters): QueryRequest => ({
+  resourceInfo,
+  queryParameters,
+  paginationParameters,
+  security: { ...newSecurity() },
+  traceId: 'tracer',
+});
+
+describe('When querying for documents', () => {
+  let client: Client;
+  let deleteDocumentAfterTest = true;
+
+  beforeAll(async () => {
+    await setupOpenSearch();
+    client = await getNewClient();
+  });
+
+  afterAll(async () => {
+    await teardownOpenSearch();
+  });
+
+  describe("given there's a document added", () => {
+    beforeEach(async () => {
+      deleteDocumentAfterTest = true;
+      await afterUpsertDocument(
+        newUpsertRequest,
+        {
+          response: 'INSERT_SUCCESS',
+        } as UpsertResult,
+        client,
+      );
+    });
+
+    afterEach(async () => {
+      if (deleteDocumentAfterTest) {
+        await afterDeleteDocumentById(
+          { id, resourceInfo } as DeleteRequest,
+          { response: 'DELETE_SUCCESS' } as DeleteResult,
+          client,
+        );
+      }
+    });
+
+    it('should be able to retrieve the document', async () => {
+      const response = await queryDocuments(setupQueryRequest({}, {}), client);
```

Comment:
What do you think about adding the following scenario?

Maybe try loading two students, one with firstName = 'Test' and one with firstName = 'Other'

Then do setup calls equivalent to the following GET requests:

* `GET students`  -> returns both
* `GET students/?Limit=1 ` -> returns only the first student uploaded (well, I'm not sure which one it will be, though it should always be the same on any system because of the `ORDER BY _doc` clause).
* `GET students/?Limit=1&Offset=1` -> returns the other student.
* `GET students/?firstName=Test` -> returns only that specific student
* `GET students/?firstName=Test&Offset=1` -> returns an empty array, since there is only one student to match
* `GET students/?firstName=\'injection+attack` -> returns nothing (I think), because we will end up searching for a blank first name instead.

Reply:
I agree, this is needed. I will add this scenario